packages/ckeditor5-restricted-editing/tests/restrictededitingmodeediting.js

```diff
@@ -476,6 +482,62 @@ describe( 'RestrictedEditingModeEditing', () => {
 			expect( markerRange.isEqual( expectedRange ) ).to.be.true;
 		} );
 
+		it( 'should extend maker when inserting inline image on the marker boundary (end)', () => {
+			setModelData( model, '<paragraph>foo bar[] baz</paragraph>' );
+			const imgSrc = 'foo/bar.jpg';
+			const firstParagraph = model.document.getRoot().getChild( 0 );
+			const command = new InsertImageCommand( editor );
+
+			model.change( writer => {
+				writer.addMarker( 'restrictedEditingException:1', {
+					range: writer.createRange( writer.createPositionAt( firstParagraph, 4 ), writer.createPositionAt( firstParagraph, 7 ) ),
+					usingOperation: true,
+					affectsData: true
+				} );
+			} );
+
+			command.execute( { source: imgSrc } );
```

Comment:
Why not `editor.execute(...)`?

Reply:
Because there is no `insertImage` command in editor. So we would need to add `Image` plugin and when we add `Image`,  tests are crash. For some reason editor created with `VirtualTestEditor` has no `ui` property so it causes this kind of errors "**_Cannot read property of undefined_**". In the code base there is only one example of usage `insertImage` with editor.**execute()** and it was used with `ClassicEditor` so final decision was to go with simple `new InsertImageCommand()`.

![image](https://github.com/ckeditor/ckeditor5/assets/47000454/19634f7e-607d-4d15-9353-2c3755140bb8)

Reply:
Unfortunately, still the same problem when I add `Image` to plugins.